app/assets/javascripts/discourse/app/components/form-template-field/upload.gjs

```diff
@@ -0,0 +1,122 @@
+import { computed } from "@ember/object";
+import { htmlSafe } from "@ember/template";
+import { tracked } from "@glimmer/tracking";
+import { dasherize } from "@ember/string";
+import { isAudio, isImage, isVideo } from "discourse/lib/uploads";
+import PickFilesButton from "discourse/components/pick-files-button";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class FormTemplateFieldUpload extends Component.extend(
+  UppyUploadMixin
+) {
+  <template>
+    <div class="control-group form-template-field" data-field-type="upload">
+      {{#if @attributes.label}}
+        <label class="form-template-field__label">
+          {{@attributes.label}}
+          {{#if @validations.required}}
+            {{icon "asterisk" class="form-template-field__required-indicator"}}
+          {{/if}}
+        </label>
+      {{/if}}
+
+      {{#if @attributes.description}}
+        <span class="form-template-field__description">
+          {{htmlSafe @attributes.description}}
+        </span>
+      {{/if}}
+
+      <input type="hidden" name={{@id}} value={{this.uploadValue}} />
+
+      <PickFilesButton
+        @fileInputClass="form-template-field__upload"
+        @fileInputId={{this.fileUploadElementId}}
+        @allowMultiple={{@attributes.allow_multiple}}
+        @showButton={{true}}
+        @onFilesPicked={{true}}
+        @icon="upload"
+        @label={{this.uploadStatusLabel}}
+        @fileInputDisabled={{this.disabled}}
+        @acceptedFormatsOverride={{@attributes.file_types}}
+        @acceptedFileTypesString={{@attributes.file_types}}
+      />
+
+      {{#if this.uploadedFiles}}
+        <ul class="form-template-field__uploaded-files">
+          {{#each this.uploadedFiles as |file|}}
+            <li>
+              {{icon "file"}}
+              <a
+                href={{file.url}}
+                target="_blank"
+                rel="noopener noreferrer"
+              >{{file.file_name}}</a>
+              <span>{{file.human_filesize}}</span>
+            </li>
+          {{/each}}
+        </ul>
+      {{/if}}
+    </div>
+  </template>
+
+  @tracked uploadValue;
+  @tracked uploadComplete = false;
+  @tracked uploadedFiles = [];
+  @tracked disabled = this.uploading;
+  @tracked fileUploadElementId = `${dasherize(this.id)}-uploader`;
+  @tracked fileInputSelector = `#${this.fileUploadElementId}`;
+
+  type = "composer";
+
+  @computed("uploading", "uploadValue")
+  get uploadStatusLabel() {
+    if (!this.uploading && !this.uploadValue) {
+      return "form_templates.upload_field.upload";
+    }
+
+    if (!this.uploading && this.uploadValue) {
+      this.uploadComplete = true;
+      return "form_templates.upload_field.upload";
+    }
+
+    return "form_templates.upload_field.uploading";
+  }
+
+  uploadDone(upload) {
+    // If re-uploading, clear the existing file if multiple aren't allowed
+    if (!this.attributes.allow_multiple && this.uploadComplete) {
```

Comment:
```suggestion
    if (!this.args.attributes.allow_multiple && this.uploadComplete) {
```

Reply:
or I suppose you can also just use the property defined in Uppy like below `!this.allowMultipleFiles`, but in any case, I don't think `this.attributes` can be accessed directly

Reply:
I agree that it shouldn't, but it is returning the proper boolean 🤔  maybe this is being handled by `ember-this-fallback`? Anyway, you're right and I've updated it.